### PR TITLE
Issue176 component diagram

### LINKED ISSUE #176: Update component diagram and add it to README
The C4 diagram needs some more updates, and should be in the README as well.

### PR DESCRIPTION
**Pull request by mortenwh**
**Summary**: closes #176 

**Related issue**:

**Suggested reviewer(s)**:

**Reviewer checklist**:

* [ ] The headers of all files contain a reference to the repository license
* [ ] 100% test coverage of new code - meaning:
  * [ ] The overall test coverage increased or remained the same as before
  * [ ] Every function is accompanied with a test suite
  * [ ] Tests are both positive (testing that the function work as intended with valid data) and negative (testing that the function behaves as expected with invalid data, e.g., that correct exceptions are thrown)
  * [ ] Functions with optional arguments have separate tests for all options
* [ ] Examples are supported by doctests
* [ ] All tests are passing
* [ ] All names (e.g., files, classes, functions, variables) are explicit
* [ ] Documentation (as docstrings) is complete and understandable

The checklist is based on the S-ENDA conventions and definition of done (see [General Conventions](https://s-enda-documentation.readthedocs.io/en/latest/general_conventions.html)). The above points are not necessarily relevant to all contributions. In that case, please add a short explanation to help the reviewer.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -4,6 +4,10 @@
 [![pytest](https://github.com/metno/discovery-metadata-catalog-ingestor/actions/workflows/pytest.yml/badge.svg?branch=main)](https://github.com/metno/discovery-metadata-catalog-ingestor/actions/workflows/pytest.yml)
 [![codecov](https://codecov.io/gh/metno/discovery-metadata-catalog-ingestor/branch/main/graph/badge.svg?token=xSG9Sg0jQ0)](https://codecov.io/gh/metno/discovery-metadata-catalog-ingestor)
 
+## Design
+
+![C4 component diagram](./dmci-component-diagram.png)
```
Comment (johtoblan) [inline README.md:9]: Maybe move it a bit lower in the README?

Comment (mortenwh) [reply README.md:9]: 1. Dependencies
2. Environment variables
3. Installation
4. Usage
5. Design
6. Tests
7. Debugging
8. License

OK? Or maybe usage first?

Comment (johtoblan) [reply README.md:9]: Think this is good

Comment (mortenwh) [reply README.md]: done

**Review by johtoblan — APPROVED** [review]
LGTM

**Pull request merged**